### PR TITLE
feat(agent-teams): run native teammate panes on Windows

### LINKED ISSUE #15751: Claude Agent Teams native panes are unreachable on Windows
## What happens

On Windows, Agent Teams always falls back to Claude Code's in-process teammates. `claude-agent-teams` is marked `detectUnsupportedRuntimes: ['win32', 'wsl']`, `buildClaudeAgentTeamsLaunchPlan` forces `in-process` on `win32`, and `orca claude-teams` refuses to start. Underneath those guards, the native-pane path is broken in three independent places.

## Why it cannot work today

**1. The `tmux` shim is unspawnable.** `ensureClaudeAgentTeamsShimDir` writes `tmux.cmd` on Windows. Claude Code's tmux backend spawns `tmux` directly with no shell, and Node refuses to spawn `.bat`/`.cmd` without `shell: true` since the CVE-2024-27980 fix (Node >= 18.20.2 / 20.12.2 / 21.7.3). The spawn fails with `EINVAL` before the shim runs, so no tmux call ever reaches `agentTeams.tmuxCompat`.

**2. Pane commands are written for `/bin/sh`.** Claude Code emits:

```
cd 'E:\repo' && env CLAUDECODE=1 CLAUDE_CODE_EXPERIMENTAL_AGENT_TEAMS=1 'C:\Users\me\.local\bin\claude.exe' --agent-id Nova@... --agent-name Nova ...
```

Real tmux runs pane commands through `/bin/sh`, so that is valid there. `ClaudeAgentTeamsTmuxDispatcher` passes it to `splitTerminal` instead, which hands it to the pane's own shell. In a PowerShell pane it dies immediately:

```
env:
Line |
 117 |  cd 'E:\repo' && env CLAUDECODE=1 CLAUDE_CODE_EXP …
     |                                       ~~~
     | The term 'env' is not recognized as a name of a cmdlet, function, script file, or executable program.
```

The pane is created and the team config records the teammate as `backendType: tmux` with a pane id, so this fails silently — the roster looks right while the pane holds a dead shell.

**3. `orca claude-teams` throws on win32.** `CORE_HANDLERS['claude-teams']` early-returns `unsupported_platform`.

## Verified

Patching all three locally on Windows 11 (Orca 1.4.x, Claude Code 2.1.238, PowerShell 7 panes) produces a working native teammate pane: the teammate runs a real Claude session in its own Orca split, answers, and reports back to the leader over team messaging.

## Known remaining issue, not covered here

Launching through the packaged `orca.exe` renders a blank TUI — running the same CLI entrypoint with plain `node` renders correctly, so something about the Electron-as-node launcher is not giving the child a usable console. That is independent of the three problems above and affects the `orca claude-teams` entrypoint only, not teams launched by Orca itself.

## Environment

- Windows 11 Pro 26200, PowerShell 7 panes
- Orca 1.4.178-rc.2 (source), installed build 1.4.x
- Claude Code 2.1.238

### PR DESCRIPTION
**Pull request by thomasytterstrom**
## ELI5

On Windows, Agent Teams quietly gave you in-process teammates instead of real Orca panes. Three separate things blocked the pane path: Orca's `tmux` shim was a `.cmd` that Claude Code cannot spawn, the pane command was written in shell syntax PowerShell does not speak, and the CLI refused to start. This fixes all three, and keeps the in-process fallback wherever the pane command cannot be spelled correctly.

## What Changed

**A shim Claude Code can actually spawn.** `OrcaCliLauncher` now enters tmux-shim mode when its own file name is `tmux` — it resolves the CLI from `ORCA_AGENT_TEAMS_SHIM_BIN` (its copy sits outside the app tree, so its own location says nothing) and prepends `agent-teams-tmux` to the argv. `ensureClaudeAgentTeamsShimDir` installs a copy of the packaged launcher as `tmux.exe` alongside the existing `tmux.cmd`. No new build artifact: the launcher is already a self-contained .NET Framework exe that forwards argv to the CLI.

**Pane commands re-spelled for the pane's shell.** New `src/shared/claude-agent-teams-pane-command.ts` translates the `cd '<dir>' && env NAME=value <argv…>` shape Claude Code writes for `/bin/sh` into `Set-Location …; $env:NAME = …; & …`, using the existing `quoteStartupArg` / `buildShellCommandFromArgv` helpers. `posix` panes are left untouched; anything it does not recognise returns null and the original text is used. `ClaudeAgentTeamsTmuxDispatcher` applies it at both pane-launch sites, using a `paneShell` now carried on the team and resolved from `terminalWindowsShell`.

**Gates lifted, with narrower ones in their place.** `orca claude-teams` no longer throws on win32, and `claude-agent-teams` drops `win32` from `detectUnsupportedRuntimes`. `buildClaudeAgentTeamsLaunchPlan` keeps the in-process fallback when the panes speak `cmd` (its `set "NAME=value"` form cannot carry a `"`, `%` or `!` safely) or when the spawnable `tmux.exe` is absent, which is the case for dev builds with no packaged launcher.

## Why

Real tmux runs pane commands through `/bin/sh`; Orca types them into the pane's own shell, so on Windows the sh-only spellings have to be translated rather than forwarded. Translating at the dispatcher keeps the tmux-compat surface honest and reuses the shell-quoting vocabulary the repo already has, instead of teaching Claude Code's command builder about Orca.

The failure this fixes was silent: the pane gets created and the team config records the teammate as `backendType: tmux` with a pane id, so the roster looks correct while the pane holds a dead shell. Degrading to in-process where the command cannot be spelled correctly keeps that from coming back in a different shell.

## Linked Issue

Fixes #15751

## Visual Proof

N/A for UI — no renderer or styling change. The behavioral proof is a teammate pane that runs instead of dying; before/after terminal output is in #15751.

## Testing

Automated:

- `src/shared/claude-agent-teams-pane-command.test.ts` — new, 8 cases: the verbatim Claude Code 2.1.238 teammate command, `posix` passthrough, `cmd` refusal, optional `cd`/`env` prefixes, apostrophe doubling, operator refusal, unbalanced-quote refusal.
- `src/main/runtime/claude-agent-teams-service.test.ts` — new case asserting `respawn-pane` reaches `splitTerminal` as PowerShell; the existing pane-lifecycle test now pins `paneShell: 'posix'` explicitly.
- `src/main/runtime/claude-agent-teams-shim-env.test.ts` — win32 now takes the native branch; new cases for the `cmd` fallback and for the missing-`tmux.exe` fallback.
- `config/scripts/build-windows-cli-launcher.test.mjs` — compiles the launcher, copies it as `tmux.exe`, and asserts argv arrives as `['agent-teams-tmux', …]`, plus a 127 exit for an unqualified `ORCA_AGENT_TEAMS_SHIM_BIN`.
- `src/cli/handlers/core.test.ts` — two cases were skipped on Windows only because of the removed `unsupported_platform` return; they now run there and pass.

Manual, on Windows 11 Pro 26200 with PowerShell 7 panes, Claude Code 2.1.238: teammate spawned into a native Orca split, ran a real Claude session, produced the correct answer, and reported back to the leader over team messaging.

`pnpm lint` and `pnpm typecheck` pass. `pnpm test` across `src/cli` and `src/shared` is clean apart from four failures that reproduce identically on an unmodified checkout of this commit (`node-markdown-document-discovery` ×2, `automation-schedules`, `external-automation-jobs-file`). `pnpm lint` also reports `resources/skills/*.json` as stale on a fresh clone, untouched here. Not tested on macOS, Linux or SSH — all changed Windows behavior is behind `process.platform` / `paneShell` checks, and the `posix` path is unchanged by construction.

- [x] I manually tested these changes locally
- [x] Automated tests added/updated, or explained why not below

## AI Disclosure

Written with Claude Opus 5 via Claude Code, from a Windows debugging session that traced the three failures. Every change was reviewed and each test run was verified by the contributor.

## Agent skill upstream boundary

- [x] Not applicable, or this change follows `docs/reference/agent-skill-sharing-upstream-boundary.md` and copies or mechanically translates no upstream skill-installer source, tests, fixtures, registry entries, path tables, comments, or documentation.

## Notes

- Security: the shim keeps the existing contract — an unqualified `ORCA_AGENT_TEAMS_SHIM_BIN` is refused with exit 127 rather than resolved against the pane cwd, and the copied `tmux.exe` resolves the CLI only from that absolute path.
- Cross-platform: macOS and Linux take the same code path as before; the translation returns null for `posix`.
- Remote SSH: `paneShell` is resolved with `isRemote: false` because teammate panes launch on the local host; remote hosts keep the previous behavior.
- Backwards compatibility: `tmux.cmd` is still written, and `paneShell` is optional with a platform default.

## Checklist

- [x] This PR is small and focused
- [x] I explained what changed and why (including ELI5)
- [x] Before/after screenshots or videos attached for UI changes, or `N/A` with reason
- [x] Self-reviewed for correctness, security, and performance
- [x] Cross-platform, SSH/remote, and path/shortcut impact considered (or N/A)
- [x] `pnpm lint`, `pnpm typecheck`, `pnpm test`, and `pnpm build` pass (or CI will cover; local preferred)

### CODE REVIEW

File: native/windows-cli-launcher/OrcaCliLauncher.cs [hunk 1]
```diff
@@ -67,6 +84,27 @@ private static int Main(string[] args)
...
+    // the team launch already published; anything unqualified is refused rather than guessed.
+    private static string ResolveTmuxShimLauncherDirectory()
+    {
+        string shimBin = Environment.GetEnvironmentVariable("ORCA_AGENT_TEAMS_SHIM_BIN");
+        if (String.IsNullOrEmpty(shimBin) || !Path.IsPathRooted(shimBin))
+        {
+            return null;
+        }
+        return Path.GetDirectoryName(shimBin);
+    }
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply native/windows-cli-launcher/OrcaCliLauncher.cs:98]: Fixed in d1d76759.

Confirmed: `Path.IsPathRooted` accepts both `\orca.exe` and `C:orca.exe`, and .NET Framework 4.0 has no `Path.IsPathFullyQualified`, so the launcher now carries an `IsFullyQualifiedPath` check that accepts only drive-absolute and UNC paths. The shim fixture in `build-windows-cli-launcher.test.mjs` now asserts exit 127 for `orca`, `\orca.exe` and `C:orca.exe`; it compiles the real launcher, so the check is exercised rather than asserted against source text.

(comment written by AI)

File: src/main/runtime/claude-agent-teams-tmux-dispatcher.ts [hunk 2]
```diff
@@ -116,7 +117,7 @@ export class ClaudeAgentTeamsTmuxDispatcher {
     const splitTarget = resolveSplitTarget(team, targetPane, parsed.flags.has('-h'))
     const split = await api.splitTerminal(splitTarget.pane.handle, {
       direction: splitTarget.direction,
-      command: parsed.positional.join(' ') || undefined,
+      command: claudeAgentTeamsPaneCommand(parsed.positional.join(' '), team.paneShell),
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply src/main/runtime/claude-agent-teams-tmux-dispatcher.ts:120]: Fixed in d1d76759, with one correction to the diagnosis.

`& 'cat'` does not exit immediately. `cat` is an alias for `Get-Content`, whose `Path` parameter is mandatory, so PowerShell blocks on a parameter prompt instead. Verified in a real Orca pane — the pane stays `running` and renders:

```
cmdlet Get-Content at command pipeline position 1
Supply values for the following parameters:
Path[0]:
```

That is why the end-to-end run worked. It is still wrong to ship: the placeholder shows the user a stray prompt, and anything typed into it would be taken as a file path to read.

The holding command now maps to `Wait-Event`, which blocks with no output and no side effects (verified in a pane: running, empty). `src/main/runtime/claude-agent-teams-service.test.ts` asserts the first `splitTerminal` command is `Wait-Event`, as suggested, and the translator test pins the mapping directly.

(comment written by AI)

**Comment by thomasytterstrom** [comment]
Addressed the nitpick too: `resolveClaudeAgentTeamsPaneShell()` is now read once per launch into `agentTeamsPaneShell` and reused by both the launch plan and the `createTeamEnv` callback.

`pnpm typecheck`, `oxlint`, the native code-quality audit and the max-lines ratchet all pass; the five affected test files pass (34 passed, 3 skipped — the skips are the two `itCrossHost` launcher cases and one POSIX-only shim test).

File: src/shared/claude-agent-teams-pane-command.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    while (tokens[0] !== undefined && ENV_ASSIGNMENT.test(tokens[0])) {
+      const pair = tokens.shift()!
+      const separator = pair.indexOf('=')
+      assignments.push({ name: pair.slice(0, separator), value: pair.slice(separator + 1) })
+    }
+  }
+  // Why: a surviving operator means sh would run something this rewrite does not
+  // model, and PowerShell would read it differently again.
+  if (tokens.length === 0 || tokens.some((token) => SH_OPERATOR.test(token))) {
+    return null
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply src/shared/claude-agent-teams-pane-command.ts]: Fixed in d22ee098.

Confirmed: `tokenizeStartupCommand` unquotes as it tokenizes, so by the time `SH_OPERATOR` ran, a quoted `|` in an argument and a bare pipe operator were the same string — `claude --prompt 'a|b'` was rejected and the pane fell back to the untranslated sh text.

The tokenizer already records the distinction the value has lost: `spans[i].divergesFromShell` is set only when the byte was read as syntax rather than as data. The rewrite now carries each token's span and rejects on that, exempting the `&&` at index 2 of a `cd` chain, which is the one operator it does model. That also closes a hole the character test left open — `$(…)` substitutions and backslash-newline continuations contain none of `;&|<>`, so they used to pass through and get flattened into a PowerShell single-quoted literal.

Two regression tests added: one asserting `cd '/repo' && env A='x|y' claude --prompt 'a|b' --filter 'c;d' --to '>e'` round-trips with every operator character intact, one asserting `claude $(id -un)` is now declined. The existing `claude | tee log` case still returns null.

`pnpm typecheck`, `oxlint` and the affected suites pass (49 passed, 1 skipped).

(comment written by AI)

File: src/main/runtime/claude-agent-teams-service.ts [hunk 4]
```diff
@@ -98,14 +103,39 @@ export class ClaudeAgentTeamsService {
     try {
       const team = this.resolveTeam(request)
       const { command, args } = splitTmuxCommand(request.argv)
-      const stdout = await this.dispatcher.dispatch(team, command, args, request.envPane, api)
+      const stdout = await this.runSerialized(team, () =>
+        this.dispatcher.dispatch(team, command, args, request.envPane, api)
+      )
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply src/main/runtime/claude-agent-teams-service.ts:109]: Fixed in a54b7251f4. `handleTmuxCompat` now re-resolves the team inside `runSerialized`, so a command that waited in the queue gets `stale or unauthorized agent team` once the leader's team is removed, instead of splitting or closing panes. New test: "does not dispatch a queued command after its team is torn down" (fails without the fix).

File: src/main/runtime/orca-runtime-resolve-terminal-split-source-authority.ts [hunk 5]
```diff
@@ -101,11 +102,66 @@ export class OrcaRuntimeWithResolveTerminalSplitSourceAuthority extends OrcaRunt
...
+    try {
+      const { leaf } = this.getLiveLeafForHandle(handle)
+      leafCoords = { tabId: leaf.tabId, leafId: leaf.leafId, paneRuntimeId: leaf.paneRuntimeId }
+    } catch {
+      leafCoords = null
+    }
+    const close = await this.closeTerminal(handle)
+    if (close.ptyKilled && leafCoords) {
+      this.notifier?.closeTerminal(leafCoords.tabId, leafCoords.paneRuntimeId)
+      await this.waitForLeafGoneFromTab(leafCoords.tabId, leafCoords.leafId)
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply src/main/runtime/orca-runtime-resolve-terminal-split-source-authority.ts]: Fixed in 69e32030e1. `closeAgentTeamsPaneTerminal` only sends the renderer close and waits for the leaf to leave the layout when `ptyKilled`, the leaf coordinates and `this.notifier` are all present; with no notifier it returns the close result immediately.

File: src/main/runtime/orca-runtime-resolve-terminal-split-source-authority.ts [hunk 6]
```diff
@@ -134,7 +190,8 @@ export class OrcaRuntimeWithResolveTerminalSplitSourceAuthority extends OrcaRunt
       leaderHandle: args.handle,
       baseEnv,
       shimDir,
-      shimBin
+      shimBin,
+      paneShell: resolveClaudeAgentTeamsPaneShell(this.store?.getSettings?.().terminalWindowsShell)
```
[the remark this answers is not in the document]
Comment (thomasytterstrom) [reply src/main/runtime/orca-runtime-resolve-terminal-split-source-authority.ts]: Fixed in 69e32030e1. The native-pane checks (pane shell, absolute CLI, `tmux.exe`) now live in one shared `resolveClaudeAgentTeamsNativeShim`, used by both the terminal launch plan and `prepareClaudeAgentTeamsLeaderForHandle`. When it returns null, `prepareLaunch` registers no team, returns only `CLAUDE_CODE_EXPERIMENTAL_AGENT_TEAMS=1`, and reports `mode: 'in-process'`; `orca claude-teams` then passes `--teammate-mode in-process`. The field is optional, so an older CLI ignores it and, with no `TMUX` in its env, Claude's auto mode still runs teammates in-process.

**Comment by thomasytterstrom** [comment]
Follow-up on the review comments that have no open thread to reply in:

- **Stale-handle close check** (`claude-agent-teams-close-until-confirmed.ts`): the accepted suggestion left an unbalanced brace, so the file did not compile. Fixed in 7bbab8e457. A stale handle now returns the last unconfirmed close (`confirmed: false`) or rethrows when there is none, and the comment matches. New test: "does not launch a replacement when the placeholder handle goes stale unconfirmed".
- **Use the selected launch plan instead of the platform** (`orca-runtime-agent-teams-launch-plan.ts`, outside diff): fixed in 69e32030e1. The launch plan now records the mode it chose, and the relaunch config follows `plan.mode` instead of forcing in-process on every Windows host. New test covers both modes on win32.
- **Preserve `cd &&` short-circuit behavior** (`claude-agent-teams-pane-command.ts`): fixed in 6c015ba32d with `Set-Location … -ErrorAction Stop` rather than `if ($?)`. A terminating error aborts the rest of the line in both Windows PowerShell 5.1 and pwsh 7, which I verified, and it keeps the one-line command shape. A Windows-only test runs the translated command in real `powershell.exe` against a missing directory and fails without the fix.
- **Long comments**: shortened to one line each in 6c015ba32d.

Also verified end to end in a dev build on Windows: `orca claude-teams` launched a leader with `--teammate-mode auto`, and a two-teammate team opened in its own split panes in the leader's tab.

**Comment by thomasytterstrom** [comment]
This is ready for review.

- Rebased onto current `main` (only one small conflict, in `orca-runtime-create-terminal.ts`, where both sides made independent edits).
- All CodeRabbit review comments are addressed; it has confirmed the fixes in the threads.
- Typecheck, the changed-files code-quality gate, and the agent-teams tests pass locally.
- Tested end to end on Windows with the rebased build: `orca claude-teams` started a leader in native-pane mode, and a two-teammate team opened each teammate in its own split pane in the leader's tab.

The CI workflows for the latest commit are waiting on approval because this PR comes from a fork. Could a maintainer approve them and take a look? Thanks!
